src/persist/clients.ts

```diff
@@ -71,15 +77,33 @@ export async function getClient(
   return clients.get(id);
 }
 
-export async function setClients(
+export function setClients(
```

Comment:
To avoid using the sync hash function I think setClient and setClients need to be removed, and their usages update to use the retrying updateClients approach.  

These are used by client gc, heartbeats and initClient.

Reply:
I agree. I realized this late last night.